### PR TITLE
added sample codes for mergehistory API documentation

### PR DESCRIPTION
**Pull request by zainababubakar**
Link to revised API documentation page in my Sandbox - https://www.mediawiki.org/wiki/User:Zaycodes/Sandbox/API:Mergehistory

### CODE REVIEW

**Comment by zainababubakar** [comment]
@brendajerop please review

**Comment by zainababubakar** [comment]
Thanks @brendajerop for your feedbacks, will work on them

**Comment by zainababubakar** [comment]
Hi @brendajerop, I've made a commit to fix the lint errors and also removed one of the files. kindly review.

**Comment by zainababubakar** [comment]
@brendajerop thanks for your time and all the feedbacks, I've made the necessary changes, kindly review.

File: php/mergehistory.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+    mergehistory.php
+
+    MediaWiki API Demos
+    Demo of `mergehistory` module: Merge the page revisions of Oldpage 
+    dating up to 2015-12-31T04:37:41Z into Newpage
+
```
Comment (brendajerop) [inline php/mergehistory.php]: Add MIT License

File: python/mergehistory.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+
+"""
+    mergehistory.py
+
+    MediaWiki API Demos
+    Demo of `mergehistory` module: Merge the page revisions of Oldpage
+    dating up to 2015-12-31T04:37:41Z into Newpage
+
```
Comment (brendajerop) [inline python/mergehistory.py]: Add MIT License

File: python/mergehistory.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+LOGIN_TOKEN = DATA['query']['tokens']['logintoken']
+
+# Step 2: Send a post request to log in using the clientlogin method.
+# import rights can't be granted using Special:BotPasswords
+# hence using bot passwords may not work.
+# See https://www.mediawiki.org/wiki/API:Login for more
+# information on log in methods.
+PARAMS_2 = {
+    "action": "login",
```
Comment (brendajerop) [inline python/mergehistory.py]: Use `clientlogin` like you've done for JavaSccript and PHP

File: javascript/mergehistory.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    mergehistory.js
+
+    MediaWiki API Demos
+    Demo of `mergehistory` module: Merge the page revisions of Oldpage
+    dating up to 2015-12-31T04:37:41Z into Newpage
+
+*/
+var request = require("request").defaults({jar: true}),
+url = "https://test.wikipedia.org/w/api.php";
+login_token = getLoginToken(); // Step 1
```
Comment (brendajerop) [inline javascript/mergehistory.js]: You've called `getLoginToken()` here and in line 92 as well. I think only one of them is necessary? You may use the one in line 92

File: javascript/mergehistory.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+loginRequest( login_token ); // Step 2
+csrftoken = getCsrfToken(); // Step 3
```
Comment (brendajerop) [inline javascript/mergehistory.js]: You've called this method twice as well

**Comment by brendajerop** [comment]
@zainababubakar One final commit to add API:Mergehistory to the readme files of Python, PHP and JavaScript folders. Mergehistory will go under `Page Operations`.

File: javascript/mergehistory.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            return;
+        }
+        var data = JSON.parse(body);
+        mergehistory(data.query.tokens.csrftoken);
+    });
+}
+
+// Step 4: Send a POST request  to merge the page revisions
+// of Oldpage dating up to 2015-12-31T04:37:41Z into Newpage
+function mergehistory(csrf_token) {
```
Comment (brendajerop) [inline javascript/mergehistory.js]: Make this `mergeHistory`

File: php/mergehistory.php [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+	$output = curl_exec( $ch );
+	curl_close( $ch );
+
+	$result = json_decode( $output, true );
+	return $result["query"]["tokens"]["csrftoken"];
+}
+
+# Step 4: Send a POST request  to merge the page revisions of Oldpage dating up to 2015-12-31T04:37:41Z into Newpage
+function mergehistory( $csrftoken ) {
```
Comment (brendajerop) [inline php/mergehistory.php]: Make this `mergeHistory`

File: php/mergehistory.php [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+  global $endPoint;
+
+  $params4 = [
+		"action"=>"mergehistory",
```
Comment (brendajerop) [inline php/mergehistory.php]: Indentation

**Comment by zainababubakar** [comment]
Hi @brendajerop, i've made all the changes

File: php/mergehistory.php [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+
+	$output = curl_exec( $ch );
+	curl_close( $ch );
+
+	$result = json_decode( $output, true );
+	return $result["query"]["tokens"]["csrftoken"];
+}
+
+# Step 4: Send a POST request  to merge the page revisions of Oldpage dating up to 2015-12-31T04:37:41Z into Newpage
+function mergeHistory( $csrftoken ) {
```
Comment (brendajerop) [inline php/mergehistory.php]: The indentation of this method doesn't seem to be consistent with the rest

Comment (zainababubakar) [reply php/mergehistory.php]: i've fixed it

Comment (brendajerop) [reply php/mergehistory.php]: @zainababubakar There's still an inconsistency in the indentation...it looks like you are using spaces in this method and tabs in the others:)

Comment (zainababubakar) [reply php/mergehistory.php]: It should be fine now, I've re-written the whole method

**Comment by brendajerop** [comment]
@zainababubakar Thank you for your work! I'm reviewing your sandbox page and I'll write my comments on it's discussion page. Meanwhile you can pick another page to work on if you wish:)

**Comment by zainababubakar** [comment]
Thanks @brendajerop

**Pull request merged**